### PR TITLE
Update rhc.erl

### PR DESCRIPTION
**Pull request by martinsumner**
Resolve dialyzer.  Also make default behaviour consistent with pb clients, in that clients will disable server_name_indication when using SSL

### CODE REVIEW

**Comment by martinsumner** [comment]
@ThomasArts - note that as well as dialyzer fixes we try and be backwards compatible (and consistent with PB API) by making server_name_indication disabled by default

File: src/rhc.erl [hunk 1]
```diff
@@ -109,14 +111,16 @@ create() ->
 %% @spec create(string(), integer(), string(), Options::list()) -> rhc()
 create(IP, Port, Prefix, Opts0) when is_list(IP), is_integer(Port),
                                      is_list(Prefix), is_list(Opts0) ->
-    Opts = case proplists:lookup(client_id, Opts0) of
-               none -> [{client_id, random_client_id()}|Opts0];
-               Bin when is_binary(Bin) ->
```
Comment (ThomasArts) [inline src/rhc.erl:114]: How could that have worked, since one would always have been a tuple returned....

File: src/rhc.erl [hunk 2]
```diff
@@ -109,14 +111,16 @@ create() ->
...
-                   [{client_id, binary_to_list(Bin)}
+    Opts = 
+        case proplists:lookup(client_id, Opts0) of
+            none ->
+                [{client_id, random_client_id()}|Opts0];
+            {client_id, Bin} when is_binary(Bin) ->
+                [{client_id, binary_to_list(Bin)}
                     | [ O || O={K,_} <- Opts0, K =/= client_id ]];
```
Comment (ThomasArts) [inline src/rhc.erl]: alternatively: `proplists:delete(client_id, Opts0)`

**Review by ThomasArts — APPROVED** [review]

File: src/rhc.erl (hunk 1 above)
Comment (martinsumner) [reply src/rhc.erl:114]: The create option with a pre-defined client_id is a legacy of very, very old riak when the vector clocks client ids would come from the application not the vnode - I think.  It is almost certainly a dead bit of code.  I guess really it should be removed

**Pull request merged**